griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,270 @@
+            saved_document_id = self._save_to_doc(save_content_params)
+            return InfoArtifact(saved_document_id)
+        except Exception as e:
+            return ErrorArtifact(f"error saving content to Google Doc with Id {document_id}: {str(e)}")
+
+    @activity(
+        config={
+            "description": "Can be used to download multiple Google Docs from Google Drive based on their paths.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_paths",
+                        description="Destination file paths on Google Drive in the POSIX format. "
+                                    "For example, 'foo/bar/baz.txt, foo/bar/baz2.txt'"
+                    ): [str]
+                }
+            ),
+        }
+    )
+    def download_google_docs(self, params: dict) -> ListArtifact | ErrorArtifact:
+        file_paths = params["file_paths"]
+        downloaded_files = []
+
+        for file_path in file_paths:
+            document_artifact = self._download_document(file_path)
+            if isinstance(document_artifact, ErrorArtifact):
+                return ErrorArtifact(f"error downloading Google Doc for path {file_path}: {document_artifact.value}")
+            downloaded_files.append(document_artifact)
+
+        return ListArtifact(downloaded_files)
+
+    def _download_document(self, file_path: str) -> BlobArtifact | ErrorArtifact:
+
+        try:
+            service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            file_id = self._path_to_file_id(service, file_path)
+            if file_id:
+                results = service.files().list(q=f"name='{file_path}'", spaces="drive", pageSize=10).execute()
+                items = results.get("files", [])
+                if items:
+                    file_id = items[0]["id"]
+                    request = service.files().export(fileId=file_id, mimeType="text/plain")
+                    response = request.execute()
+                    return BlobArtifact(response)
+                else:
+                    return ErrorArtifact(f"Error: File not found for path {file_path}")
+            else:
+                return ErrorArtifact(f"Error: File not found for path {file_path}")
+    
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error downloading Google Doc: {e}")
+
+    def _build_client(self, scopes: list[str], version: str, tool: str) -> Resource:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+        delegated_credentials = credentials.with_subject(self.owner_email)
+        service = build(tool, version, credentials=delegated_credentials)
+        return service
+
+    def _path_to_file_id(self, service, path: str) -> Optional[str]:
```

Comment:
Isn't this the same code you used in the Google Drive tool? Why repeat it here?

Reply:
Ok, i can just inherit the drive class!